### PR TITLE
blacklist Openblas 0.2.16 for missing _dggsvd_ and _sggsvd_

### PR DESCRIPTION
**Pull request by davydden**
fixes https://github.com/LLNL/spack/issues/621

### CODE REVIEW

**Comment by eschnett** [comment]
Is it necessary to blacklist this version entirely? Instead, Trilinos could blacklist it in its dependency specification, keeping this OpenBLAS version available for others.

**Comment by tgamblin** [comment]
How about just adding preferred=True to the older one?

**Comment by eschnett** [comment]
I just find <https://github.com/scipy/scipy/issues/5266>: `dggsvd` is deprecated, and won't come back. Instead, there is e.g. `dggsvd3`.

This probably means that this is intended in OpenBLAS.

**Comment by eschnett** [comment]
See <http://www.openblas.net/Changelog.txt>:
```
	  Add BUILD_LAPACK_DEPRECATED option in Makefile.rule to build
	  LAPACK deprecated functions.
```
It seems we could add this as a variant, maybe even enabled by default.

**Comment by eschnett** [comment]
The saga continues. Apparently, OpenBLAS 0.2.17 was recently released, with this changelog entry:
```
	* Enable BUILD_LAPACK_DEPRECATED=1 by default.
```
So all we need to do is add this new version.

File: var/spack/repos/builtin/packages/openblas/package.py [hunk 1]
```diff
@@ -6,15 +6,20 @@ class Openblas(Package):
     homepage = "http://www.openblas.net"
     url      = "http://github.com/xianyi/OpenBLAS/archive/v0.2.15.tar.gz"
 
-    version('0.2.16', 'fef46ab92463bdbb1479dcec594ef6dc')
+    # version('0.2.16', 'fef46ab92463bdbb1479dcec594ef6dc') # blacklist: miss _dggsvd_ and _sggsvd_ at least on OSX
     version('0.2.15', 'b1190f3d3471685f17cfd1ec1d252ac9')
 
+    variant('tests', default=True, description='Run small tests')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/openblas/package.py]: @davydden Why a variant (that changes the hash) to run `make test`?

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openblas/package.py]: We might want to leave this out for now and wait until a consensus is reached in #169.

Comment (davydden) [inline var/spack/repos/builtin/packages/openblas/package.py]: @alalazo i think it's always good to run tests, especially if they are (i) not time consuming (ii) serial.

Comment (alalazo) [inline var/spack/repos/builtin/packages/openblas/package.py]: @davydden I completely agree on tests, but why with a variant?

Comment (davydden) [inline var/spack/repos/builtin/packages/openblas/package.py]: @alalazo right, i can remove it :smile:  will do.

**Comment by davydden** [comment]
i updated the PR with the new version and a fix for 0.2.16

**Pull request merged**